packages/inter-protocol/src/psm/psmCharter.js

```diff
@@ -16,80 +18,114 @@ import { E, Far } from '@endo/far';
+      fit(args, paramChangesOfferArgsShape);
+      seat.exit();
+
+      const {
+        params,
+        instance,
+        deadline,
+        path = { paramPath: { key: 'governedApi' } },
+      } = args;
+      const psmGovernor = instanceToGovernor.get(instance);
+      return E(psmGovernor).voteOnParamChanges(counter, deadline, {
         ...path,
         changes: params,
       });
     };
 
-    return zcf.makeInvitation(
-      () => Far('param change hook', { voteOnParamChanges }),
-      'vote on param changes',
-    );
+    return zcf.makeInvitation(voteOnParamChanges, 'vote on param changes');
   };
 
-  /** @param {Instance} instance */
-  const makeOfferFilterInvitation = instance => {
-    const psm = instanceToCreator.get(instance);
-    /**
-     * @param {string[]} strings
-     * @param {bigint} deadline
-     */
-    const voteOnOfferFilter = (strings, deadline) => {
-      return E(psm).voteOnOfferFilter(counter, deadline, strings);
+  const makeOfferFilterInvitation = (instance, strings, deadline) => {
+    const voteOnOfferFilterHandler = seat => {
+      seat.exit();
+
+      const psmGovernor = instanceToGovernor.get(instance);
+      return E(psmGovernor).voteOnOfferFilter(counter, deadline, strings);
     };
 
-    return zcf.makeInvitation(
-      () => Far('offer filter hook', { voteOnOfferFilter }),
-      'vote on offer filter',
-    );
+    return zcf.makeInvitation(voteOnOfferFilterHandler, 'vote on offer filter');
   };
 
-  const invitationMakers = Far('PSM Invitation Makers', {
-    VoteOnParamChange: makeParamInvitaion,
-    VoteOnPauseOffers: makeOfferFilterInvitation,
+  const makerShape = M.interface('PSM Charter InvitationMakers', {
+    VoteOnParamChange: M.call().returns(M.promise()),
+    VoteOnPauseOffers: M.call(
+      InstanceHandleShape,
+      M.arrayOf(M.string()),
+      TimestampShape,
+    ).returns(M.promise()),
   });
+  const invitationMakers = makeHeapFarInstance(
+    'PSM Invitation Makers',
+    makerShape,
+    {
+      VoteOnParamChange: makeParamInvitation,
+      VoteOnPauseOffers: makeOfferFilterInvitation,
+    },
+  );
 
   const charterMemberHandler = seat => {
     seat.exit();
-    return invitationMakers;
+    return harden({ invitationMakers });
   };
 
-  const creatorFacet = Far('psm charter creator', {
-    /**
-     * @param {Instance} psmInstance
-     * @param {GovernedContractFacetAccess<{},{}>} psmCreatorFacet
-     * @param {Brand} [anchor] for diagnostic use only
-     * @param {Brand} [minted] for diagnostic use only
-     */
-    addInstance: (psmInstance, psmCreatorFacet, anchor, minted) => {
-      console.log('psmCharter: adding instance', { minted, anchor });
-      instanceToCreator.init(psmInstance, psmCreatorFacet);
-    },
-    makeCharterMemberInvitation: () =>
-      zcf.makeInvitation(charterMemberHandler, 'PSM charter member invitation'),
+  const psmCharterCreatorI = M.interface('PSM Charter creatorFacet', {
```

Comment:
nit: this is used only once so better to inline, imo

Reply:
I see it as part of the declaration of the instance and that putting it outside clutters following the logic and adds cognitive load to track what is used where.

we should come up with a shared coding style for this. I'm willing to disagree and commit so we can be consistent.